### PR TITLE
Testing domain PA fix and refactor

### PR DESCRIPTION
**Pull request by gaiksaya**
*Issue #, if available:*
https://github.com/opendistro-for-elasticsearch/opendistro-infra/projects/8

*Description of changes:*

1. Add endpoint for PA on port 9600
2. Create new LB with port 9600 manually and map it to the appropriate target group
3. Modify the ES/Kibana installation (userdata script) to expose the PA accessible outside the cluster
4. Change AMI with recent ones
5. Refactor userdata.sh
6. Refactor set-up workflow to remove tmp files

*Test Results:*
All the testing domains are active. Verified working of all.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by peterzhuamazon — APPROVED** [review]

File: .github/scripts/userdata.sh [hunk 1]
```diff
@@ -152,9 +160,11 @@ sudo sed -i 's/https/http/' /etc/kibana/kibana.yml
 EOF
 else
 sed -i "s/^echo \"cluster.name.*/echo \"cluster.name \: odfe-$ODFE_VER-$1-noauth\" \>\> config\/elasticsearch.yml/g" $REPO_ROOT/userdata_$1.sh
-sed -i "/echo \"network.host/a echo \"opendistro_security.disabled: true\" \>\> config\/elasticsearch.yml" $REPO_ROOT/userdata_$1.sh
 cat <<- EOF >> userdata_$1.sh
 sudo rm -rf plugins/opendistro_security
+sed -i /^opendistro_security/d config/elasticsearch.yml
+cd /opendistroforelasticsearch-kibana/
+sudo rm -rf plugins/opendistro_security
```
Comment (camerski) [inline .github/scripts/userdata.sh:167]: Duplicate of line 164?

File: .github/templates/odfe-testing-cluster-static-domain-template.json [hunk 2]
```diff
@@ -36,21 +36,25 @@
...
     },
 
     "Mappings" : {
         "DistributionMap" : {
-          "RPM" : { "amiId" : "ami-0e34e7b9ca0ace12d" },
-          "TAR" : { "amiId" : "ami-003634241a8fcdec0" },
-          "DEB" : { "amiId" : "ami-003634241a8fcdec0" }
+          "RPM" : { "amiId" : "ami-01ce4793a2f45922e" },
+          "TAR" : { "amiId" : "ami-07a29e5e945228fa1" },
+          "DEB" : { "amiId" : "ami-07a29e5e945228fa1" }
```
Comment (camerski) [inline .github/templates/odfe-testing-cluster-static-domain-template.json:57]: What is the difference between these AMIs? It is surprising that tar and deb share an AMI, but rpm does not.

Adding a comment explaining the difference would be super helpful.

File: .github/scripts/userdata.sh (hunk 1 above)
Comment (peterzhuamazon) [reply .github/scripts/userdata.sh:167]: Good catch.

Comment (gaiksaya) [reply .github/scripts/userdata.sh:167]: One is removing from elasticsearch folder and other from kibana plugins' folder

File: .github/templates/odfe-testing-cluster-static-domain-template.json (hunk 2 above)
Comment (gaiksaya) [reply .github/templates/odfe-testing-cluster-static-domain-template.json:57]: RPM- Amazon Linux Machine 
DEB/RPM - Ubuntu
I wish I can add comments but its JSON :(

Comment (peterzhuamazon) [reply .github/templates/odfe-testing-cluster-static-domain-template.json:57]: Not surprising at all, TAR just need a generic linux so sharing the same ubuntu image is ok.
As for the changes, @gaiksaya was initially using AMI provided by community, which has /dev/shm as 755, thus cause issues with PA writing to the path with non-root user. We changed it to the official Ubuntu/AL2 images provided by AWS, which have 777 for /dev/shm by default here.

File: .github/scripts/userdata.sh (hunk 1 above)
Comment (camerski) [reply .github/scripts/userdata.sh:167]: Ah, I see. Thanks!

File: .github/templates/odfe-testing-cluster-static-domain-template.json (hunk 2 above)
Comment (camerski) [reply .github/templates/odfe-testing-cluster-static-domain-template.json:57]: Oh yeah, JSON :(

Thanks for the explanation.

**Review by camerski — APPROVED** [review]

**Pull request merged**